scripts/install.sh

```diff
@@ -1,159 +1,293 @@
+          ;;
+      esac
       ;;
-    n) DISABLE_NETWORKING="true"
+    n | no-networking) DISABLE_NETWORKING="true"
       ;;
-    q) QUIET="true"
+    q | quiet) QUIET="true"
+      ;;
+    \?)  # Handle invalid short options
+      die "Error: Invalid option -$OPTARG" \
+          "See './install.sh -h' for more information."
+      ;;
+    * )  # Handle invalid long options
+      die "Error: Invalid option --$OPT" \
+          "See './install.sh -h' for more information."
       ;;
-    \?)
-      echo "Error: Invalid option -- '$OPTARG'"
-      echo "Try './install.sh -h' for more information."
-      exit 1
   esac
 done
 
-shift $(($OPTIND -1))
-
 if [ "$(id -u)" != "0" ]; then
-   echo "This script must be run as root" 1>&2
-   exit 1
+   die "This script must be run as root"
+fi
+
+if [[ -z "$ARCH" ]]; then
+  debug "Arch was not specified. Inferring..."
+  ARCH=$(uname -m)
+  debug "Arch was inferred to be $ARCH"
 fi
 
-ARCH=$(uname -m)
 ARCH_NAME=""
 if [ "$ARCH" = "aarch64" ]; then
   ARCH_NAME="linuxarm64"
 elif [ "$ARCH" = "armv7l" ]; then
-  echo "ARM32 is not supported by PhotonVision. Exiting."
-  exit 1
+  die "ARM32 is not supported by PhotonVision. Exiting."
 elif [ "$ARCH" = "x86_64" ]; then
   ARCH_NAME="linuxx64"
 else
-  if [ "$#" -ne 1 ]; then
-      echo "Can't determine current arch; please provide it (one of):"
-      echo ""
-      echo "- linuxarm64 (64-bit Linux ARM)"
-      echo "- linuxx64   (64-bit Linux)"
-      exit 1
-  else
-    echo "Can't detect arch (got $ARCH) -- using user-provided $1"
-    ARCH_NAME=$1
-  fi
+  die "Unsupported or unknown architecture: '$ARCH'." \
+  "Please specify your architecture using: ./install.sh -a <arch> " \
+  "Run './install.sh -h' for more information."
 fi
 
-echo "This is the installation script for PhotonVision."
-echo "Installing for platform $ARCH_NAME"
+debug "This is the installation script for PhotonVision."
+debug "Installing for platform $ARCH"
+
+# Quiet makes the default "no" instead of "ask".
+if [[ -n "$QUIET" && "$INSTALL_NETWORK_MANAGER" == "ask" ]] ; then
+  INSTALL_NETWORK_MANAGER="no"
+fi
 
 DISTRO=$(lsb_release -is)
-if [[ "$DISTRO" = "Ubuntu" && "$INSTALL_NETWORK_MANAGER" != "true" && -z "$QUIET" && -z "$DISABLE_NETWORKING" ]]; then
-  echo ""
-  echo "Photonvision uses NetworkManager to control networking on your device."
+
+# NetworkManager cannot be installed on non-Ubuntu distros
+# --no-networking takes precedence over --install-nm
+if [[ "$DISTRO" != "Ubuntu" || -n "$DISABLE_NETWORKING" ]] ; then
+  INSTALL_NETWORK_MANAGER="no"
+fi
+
+if [[ "$INSTALL_NETWORK_MANAGER" == "ask" ]]; then
+  debug ""
+  debug "Photonvision uses NetworkManager to control networking on your device."
   read -p "Do you want this script to install and configure NetworkManager? [y/N]: " response
   if [[ $response == [yY] || $response == [yY][eE][sS] ]]; then
-    INSTALL_NETWORK_MANAGER="true"
+    INSTALL_NETWORK_MANAGER="yes"
   fi
 fi
```

Comment:
I still don't think this matches the original logic. The test for Ubuntu was intended to ask for permission if someone was installing it on Ubuntu to make sure that it wasn't messing up their network configuration. This logic will override `--install-nm=yes` if the platform isn't Ubuntu, even though the user explicitly calls for it to be installed. I'll propose this alternative:

```suggestion
DISTRO=$(lsb_release -is)

if [[ "$INSTALL_NETWORK_MANAGER" == "ask" ]]; then
  if [[ "$DISTRO" != "Ubuntu" || -n "$DISABLE_NETWORKING" || -n "$QUIET" ]] ; then
    # Only ask if it makes sense to do so
    INSTALL_NETWORK_MANAGER="no"
  else
    debug ""
    debug "Photonvision uses NetworkManager to control networking on your device."
    read -p "Do you want this script to install and configure NetworkManager? [y/N]: " response
    if [[ $response == [yY] || $response == [yY][eE][sS] ]]; then
      INSTALL_NETWORK_MANAGER="yes"
    else
      INSTALL_NETWORK_MANAGER="no"
    fi
  fi
fi
```

Reply:
Yeah I can make these changes.  I had the impression that NetworkManager was only available on Ubuntu which is why I had that logic.

Reply:
Not a huge fan of large nested ifs. I think it's more readable just to have two seperate ones, but the behavior is equivalent. You can see the most recent commit for that.

Reply:
Thank you. This looks right to me.